### ISSUE TITLE
GSLB-Servers-VirtualServers - add ability to name each virtual server

### ISSUE THREAD

**Issue by LeonardosGitHub** [F5Networks/f5-appsvcs-extension#168]
##### Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

##### ISSUE TYPE

 - Feature Idea

##### AS3 BUILD/ VERSION

```
3.12.0-5
```

##### BIGIP VERSION

```
14.0.0.1
```

##### SUMMARY

```
GSLB-Servers-VirtualServers - add ability to provide a name for each virtual server.  Currently AS3 automatically names each virtual server numerically starting with 0. I'd like to ability to provide a name which allow me to write a pool configuration to reference the name inputted with the declaration.
```

##### STEPS TO REPRODUCE

```
{
    "class": "ADC",
    "schemaVersion": "3.11.0",
    "Common": {
        "class": "Tenant",
        "label": "commonTenant",
        "remark": "Declaration Common tenant",
        "enable": true,
        "Shared": {
            "class": "Application",
            "template": "shared",
            "enable": true,
            "testServer": {
                "class": "GSLB_Server",
                "dataCenter": {
                    "bigip": "/Common/test"
                },
                "devices": [
                    {
                        "address": "10.1.1.1"
                    }
                ],
                "virtualServers": [
                    {
                        "address": "10.2.2.2",
                        "port": 5050
                    },
                    {
                        "address": "10.3.3.3",
                        "port": 5051
                    }
                ]
            }
        }
    }
}
```

##### EXPECTED RESULTS

```
I would like AS3 to provide the ability to name the virtual servers, example below.

list gtm server:
gtm server testServer {
    addresses {
        10.1.1.1 {
            device-name 0
        }
    }
    datacenter test
    devices {
        0 {
            addresses {
                10.1.1.1 { }
            }
        }
    }
    metadata {
        as3 { }
    }
    monitor bigip 
    product bigip
    virtual-servers {
        virtualServerName {
            destination 10.2.2.2:5050
        }
        virtualServerNextName {
            destination 10.3.3.3:5051
        }
    }
}
```

##### ACTUAL RESULTS

```
This is what AS3 currently builds, the name of the virtual servers below are "0" and "1". 

list gtm server:
gtm server testServer {
    addresses {
        10.1.1.1 {
            device-name 0
        }
    }
    datacenter test
    devices {
        0 {
            addresses {
                10.1.1.1 { }
            }
        }
    }
    metadata {
        as3 { }
    }
    monitor bigip 
    product bigip
    virtual-servers {
        0 {
            destination 10.2.2.2:mmcc
        }
        1 {
            destination 10.3.3.3:ita-agent
        }
    }
}
```

**Comment by dstokesf5** [comment]
Being able to change the name of a GSLB virtual server is already being tracked in our internal product backlog as AUTOTOOL-207.

**Comment by dstokesf5** [comment]
This was resolved in the 3.21.0 release.
